### ISSUE TITLE
Python SDK for KServe and Kubeflow Pipelines can not be installed at the same time

### ISSUE THREAD

**Issue by thesuperzapper** [kserve/kserve#3730]
### What happened?

Currently, If you try install the latest versions of the `kfp` and `kserve` libraries at the same time, there is a dependency version conflict.

This is caused by a the following conflict:

1. [`kfp==2.7.0` needs `protobuf>=4.21.1,<5`](https://github.com/kubeflow/pipelines/blob/sdk-2.7.0/sdk/python/requirements.in#L21) but [`kserve` needs `protobuf=^3.19.0`](https://github.com/kserve/kserve/blob/v0.13.0/python/kserve/pyproject.toml#L47)

See for yourself by trying the following command:

```shell
pip install kfp==2.7.0 kserve==0.13.0
```

For those wondering, the latest versions which will work together are:

```shell
pip install kfp==2.5.0 kserve==0.13.0
```

### What did you expect to happen?

The latest versions of both `kfp` and `kserve` can be installed at the same time.

**Comment by thesuperzapper** [comment]
I have raised a similar issue for Katib, which also has a dependency conflict with `protobuf`: https://github.com/kubeflow/katib/issues/2346

**Comment by sivanantha321** [comment]
We already working on upgrading the protobuf🙃.

**Comment by sivanantha321** [comment]
will be resolved by #3751

**Comment by kimwnasptd** [comment]
Thanks for the context @sivanantha321! @yuzisun @sivanantha321 @andyi2it is there any progress on this one? Some of our users are also pinging us about not being able to use the SDKs together of the 1.9 release
